reazon.el

```diff
@@ -164,25 +166,30 @@ STREAM-2, else append them as usual."
           stream-2
           (funcall rest)))))))
 
-(defun reazon--pull (stream)
-  "Force STREAM until it isn't a suspension, then return it."
+(defun reazon--pull (stream &optional stop-time)
+  "Force STREAM until it isn't a suspension or STOP-TIME is reached, then return it."
   (let ((result stream))
     (while (functionp result)
+      (when (and stop-time (< stop-time (float-time)))
```

Comment:
Does this really need to throw an error? If the timeout is reached, then it will just return whatever it's gotten so far, right? Maybe it could just break the `while` loop here and return.

On the other hand, the user might want to know if their query has stopped because it reached the end or because it hit a timeout and there was still more to do.